### ISSUE TITLE
Regex for uuid with question marks doesn't work

### ISSUE THREAD

**Issue by wafcio** [sds/mock_redis#178]
I have used `????????-????-????-????-????????????` regex to check UUID:

```
scan_each(match: "foo/????????-????-????-????-????????????")
```
unfortunately it doesn't find key like `foo/25892691-7ec6-4b55-8ce1-341315736029`

I have made a small investigation of this issue and found https://github.com/sds/mock_redis/blob/master/lib/mock_redis/database.rb#L315
```
> "????????-????-????-????-????????????".gsub(/([^\\])\?/, '\\1.')
=> "?.?.?.?.-.?.?-.?.?-.?.?-.?.?.?.?.?.?"
```
which is IMHO wrong ruby regexp for UUID.

I have checked `????????-????-????-????-????????????` in real Redis and it works.

**Comment by sds** [comment]
Hey @wafcio, thanks for the report! 

Would welcome a pull request with accompanying test which fixes this issue. Thank you!

**Comment by sds** [comment]
Fixed in #186.
